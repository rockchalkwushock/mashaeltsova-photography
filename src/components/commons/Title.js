import React from 'react'
import styled from 'styled-components'

export const StyledTitle = styled.h1`
  color: ${({ color }) => color};
`

StyledTitle.displayName = 'StyledTitle'
StyledTitle.defaultProps = {
  color: 'lightcoral'
}

const Title = ({ color, text }) => (
  <StyledTitle color={color}>{text}</StyledTitle>
)

export default Title
