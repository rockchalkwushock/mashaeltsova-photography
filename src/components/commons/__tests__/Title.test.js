import React from 'react'
import { mount, shallow } from 'enzyme'

import Title, { StyledTitle } from '../Title'

describe('Component: <Title />', () => {
  test('should render markup', () => {
    const wrapper = mount(<Title text="Hello World!" />)
    expect(wrapper).toMatchSnapshot()
  })
  test('should have props', () => {
    const wrapper = shallow(<Title color="blue" text="Hello World!" />)
    expect(wrapper.props()).toHaveProperty('children', 'Hello World!')
    expect(wrapper.props()).toHaveProperty('color', 'blue')
  })
})

describe('Styled Component: <StyledTitle />', () => {
  test('should render markup as default', () => {
    const wrapper = shallow(<StyledTitle />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper).toHaveStyleRule('color', 'lightcoral')
  })
  test('should render markup with props', () => {
    const wrapper = shallow(<StyledTitle color="blue" />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper).toHaveStyleRule('color', 'blue')
  })
})
