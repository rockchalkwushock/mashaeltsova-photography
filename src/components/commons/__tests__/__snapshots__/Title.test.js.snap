// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Title /> should render markup 1`] = `
.c0 {
  color: lightcoral;
}

<Title
  text="Hello World!"
>
  <StyledTitle
    color="lightcoral"
  >
    <h1
      className="c0"
      color="lightcoral"
    >
      Hello World!
    </h1>
  </StyledTitle>
</Title>
`;

exports[`Styled Component: <StyledTitle /> should render markup as default 1`] = `
.c0 {
  color: lightcoral;
}

<h1
  className="c0"
  color="lightcoral"
/>
`;

exports[`Styled Component: <StyledTitle /> should render markup with props 1`] = `
.c0 {
  color: blue;
}

<h1
  className="c0"
  color="blue"
/>
`;
